mlflow/utils/search_utils.py

```diff
@@ -558,6 +584,48 @@ def paginate(cls, runs, page_token, max_results):
     VALID_SEARCH_KEYS_FOR_MODEL_VERSIONS = set(["name", "run_id", "source_path"])
     VALID_SEARCH_KEYS_FOR_REGISTERED_MODELS = set(["name"])
 
+    @classmethod
+    def _is_valid_identifier_list(cls, value_token):
+        if len(value_token._groupable_tokens) == 0:
+            raise MlflowException(
+                "While parsing a list in the query,"
+                " expected a non-empty list of string values, but got empty list",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        elif not isinstance(value_token._groupable_tokens[0], IdentifierList):
```

Comment:
New from after our conversation, since we need to ensure that the first element is actually be a list (might not in the case of a list that looks like `(,)`)